### DISCUSSION TITLE
Contract read actions with Foundry/Anvil

### DISCUSSION THREAD

**Discussion by davebryson** [wevm/viem#379] [Question] [answered]
How do you get access to read actions on a contract  with `foundry` as the chain?  

I can create a WalletClient which gives me contract write actions, but how do you get read actions?  I tried  creating a public client with foundry set as the chain, but passing it to getContract results in a `is not assignable to parameter of type 'GetContractParameters...` error.

**Comment by tmm** [comment, marked as answer]
Looks like a TypeScript error. Can you share the full error and a minimal reproducible example?

**Reply by davebryson** [reply]
Well actually, after creating a separate small project to try to reproduce the error, the problem went away.  All is working as expected.  It looks like the problem was me. :-) Thanks for the response.
